BUG:warn on Nan in minimum,maximum for scalars, float16

mattip opened the pull request:
Fixes #11589 

PR #11043 added a check for Nan in the minimum, maximum loops.

Scalars and `float16` take a different code path through maximum, minimum that was not checked. Tests added that failed before and pass after the fix.

mattip committed:
BUG:warn on Nan in minimum,maximum for scalars

tylerjereddy reviewed (commented):
Single CI test failure on posix and windows involving `np.maximum` and `float16` is probably related.

mattip commented:
Removed code that emits a warning on `float16`, if desired that should be a separate PR

seberg commented:
The float16 warnings still seem to show up in the tests?

seberg commented on numpy/core/tests/test_umath.py:
```diff
@@ -1339,6 +1339,12 @@ def test_reduce_warns(self):
+        with suppress_warnings() as sup:
+            sup.record(RuntimeWarning)
+            np.minimum(np.nan, 1)
+
```
`assert_warns` should be fine here? But the test does not actually check that the one warning is recorded like this.

charris commented:
What was the problem with float16?

mattip commented:
`float16` never emit a `npy_set_floatstatus_invalid` in `np.maximum`, `np.minimum` even if Nan appears, unlike other floats. In the original PR I added `npy_set_floatstatus_invalid` to the HALF_{maximum, minimum} loops, but this goes beyond the original issue and changes current behavior, so I removed it in a subsequent changeset

charris commented:
@mattip Sounds like float16 should raise a warning just to be consistent, what is the argument against that?

seberg commented:
I thought the argument was: "lets not go there for a minor release"? then again, it might have been, lets put in the quick bug fix before the release happens at the time.

I definitely agree that we should enable the warnings for float16, but I would say that now that 15.0 is out, it is simplest to just wait for the next release for the float16 part (I assume the float32 is partially a regression here), it is not a huge deal and has been buggy forever.

mattip commented:
See PR #11660 which extends this for 1.16 (no backport of that PR needed)
